dulwich/repo.py

```diff
@@ -676,10 +677,17 @@ def __init__(self, root):
             raise NotGitRepository(
                 "No git repository was found at %(path)s" % dict(path=root)
             )
+        worktree_commondir = os.path.join(self._controldir, COMMONDIR)
+        if(os.path.isfile(worktree_commondir)):
```

Comment:
Please just try to open this file, rather than probing for it ("ask forgiveness, not permission"). That saves a syscall per repo open.